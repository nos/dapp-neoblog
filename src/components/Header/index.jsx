/**
 * IMPORTS:
 * react
 * react-jss
 * prop-types
 */
import React from "react";
import injectSheet from "react-jss";
import PropTypes from "prop-types";

/**
 * COMPONENTS:
 * SpinningLogo
 */
import SpinningLogo from "./../SpinningLogo";

const styles = {
  header: {
    backgroundColor: "#11242D",
    color: "white",
    padding: "24px",
    marginBottom: "32px"
  },
  title: {
    fontSize: "1.5em"
  }
};

const Header = ({ classes, title }) => (
  <header className={classes.header}>
    <h1 className={classes.title}>{title}</h1>
    <SpinningLogo />
  </header>
);

/**
 * PROPTYPES:
 * classes: object
 * title: string
 */
Header.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired
};

export default injectSheet(styles)(Header);
