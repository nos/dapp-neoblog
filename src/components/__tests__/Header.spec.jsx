import React from "react";
import { render } from "enzyme";

import TestPropTypes from "./../../__helpers__/TestPropTypes.helper";

import Header from "./../Header";

describe("Components", () => {
  describe("Header", () => {
    test("should render", () => {
      const wrapper = render(<Header title="Hello" subTitle="world!" />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should validate props", () => {
      TestPropTypes(<Header title="Hello" subTitle="world!" />, 0);
      TestPropTypes(<Header />, 2);
    });
  });
});
