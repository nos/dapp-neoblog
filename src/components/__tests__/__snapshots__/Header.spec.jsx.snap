// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header should render 1`] = `
<header
  class="Header-header-0-1-1"
>
  <h1
    class="Header-title-0-1-2"
  >
    Hello world!
  </h1>
  <div
    class="SpinningLogo-logoContainer-0-1-3"
  >
    <img
      alt="parcelBack"
      class="SpinningLogo-parcelBack-0-1-4"
      src="test-file-stub"
    />
    <img
      alt="parcelReact"
      class="SpinningLogo-parcelReact-0-1-5"
      src="test-file-stub"
    />
    <img
      alt="parcelFront"
      class="SpinningLogo-parcelFront-0-1-6"
      src="test-file-stub"
    />
  </div>
</header>
`;
